docs/proposals/202005_scalable-rule-storage.md

```diff
@@ -0,0 +1,68 @@
+---
+title: Scalable rule storage
+type: proposal
+menu: proposals
+status: proposed
+---
+
+## Summary
+
+There is no way to scale rule evaluation and storage today except functionally
+sharding rules onto multiple instances the `thanos rule` component. However, we
+have already solved scaling storage of time-series across multiple processes:
+`thanos receive`.
+
+To scale storing results of rule evaluations this proposal proposes to allow
+the `thanos rule` component to have a stateless mode, storing results of
+queries by sending it to a `thanos receive` hashring instead of storing them
+locally.
+
+## Motivation
+
+A few large rules can create a significant amount of resulting time-series,
+which should not be limited to what we can scale to with a single TSDB embedded
+in the rule component.
```

Comment:
```suggestion
in the rule component.

Additionally, scaling out Ruler in terms of rule evaluations causes further defragmentation of TSDB blocks - as many of rules we have, that many of TSDB "streams" we have. While doable with vertical compaction, it might cause some operational complexity and unnecessary load on the system.
```

Reply:
Added some of this but rewrote it slightly, could you double check if that's what you were trying to say?

Reply:
LGTM! thanks